base/file.jl

```diff
@@ -47,7 +47,19 @@ function mkpath(path::AbstractString, mode::Unsigned=0o777)
     dir = dirname(path)
     (path == dir || isdir(path)) && return
     mkpath(dir, mode)
-    mkdir(path, mode)
+    try
+        mkdir(path, mode)
+    # If there is a problem with making the directory, but the directory
+    # does in fact exist, then ignore the error. Else re-throw it.
+    catch err
+        # isdir comes first because I figure it will return false more often
+        # than isa
```

Comment:
I couldn't figure out an instance where that would happen. Of course that's no guarantee. That being said, which of the following is our desired behavior:

 - Suppress the "file exists" system error and continue making new directories
 - or, suppress any error as long as directories continue to be created